### PR TITLE
Switch job dependency type to OrderOnce for material assets when property names are present.

### PR DESCRIPTION
**Pull request by hershey5045**
Notes:
- When includeMaterialPropertyNames is enabled, job dependency types for all material files can be set to process only once. Then if the materialtype file is changed in the future, we can rely on the newly implemented load-time mechanism to add the new properties. 
- Remove source dependency in MaterialBuilder. We only need the job dependency. 
- AP load time when adding a dummy property in StandardPBR.materialtype is down from 45s to 10s. 

Testing:
- Observed that adding a dummy property in StandardPBR.materialtype does not cause fbx and standalone material asset rebuild.
- Added a dummy property in StandardPBR.materialtype then reprocessed the fbx and standalone material asset. Observed that removing the dummy property does not cause a rebuild of the fbx and standalone material asset.

### CODE REVIEW

File: Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp [hunk 1]
```diff
@@ -93,8 +94,12 @@ namespace AZ
                     {
                         AssetBuilderSDK::JobDependency jobDependency;
                         jobDependency.m_jobKey = jobKey;
-                        jobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
                         jobDependency.m_sourceFile.m_sourceFileDependencyPath = file;
+
+                        bool includeMaterialPropertyNames = true;
+                        RPI::MaterialConverterBus::BroadcastResult(includeMaterialPropertyNames, &RPI::MaterialConverterBus::Events::ShouldIncludeMaterialPropertyNames);
```
Comment (santorac) [inline Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp:100]: This setting should not be used here. MaterialConverter is only part of the SceneAPI build pipeline, for converting FBX (etc) materials to Atom materials. Notice the settings appear under "/O3DE/SceneAPI/MaterialConverter", they are specific to SceneAPI.

I think we can either keep the default of includePropertyNames=true and don't make it configurable for now, or add a new setreg section for MaterialBuilder settings (in that case I suggest "/O3DE/Atom/RPI/MaterialBuilder/IncludeMaterialPropertyNames").

Comment (santorac) [reply Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp:100]: We should probably have a comment to explain why we are using OrderOnce.

File: Gems/Atom/RPI/Code/Source/RPI.Builders/Model/MaterialAssetBuilderComponent.cpp [hunk 2]
```diff
@@ -90,7 +87,10 @@ namespace AZ
                 jobDependency.m_jobKey = "Atom Material Builder";
                 jobDependency.m_sourceFile = materialTypeSource;
                 jobDependency.m_platformIdentifier = platformIdentifier;
-                jobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
+
+                bool includeMaterialPropertyNames = true;
+                RPI::MaterialConverterBus::BroadcastResult(includeMaterialPropertyNames, &RPI::MaterialConverterBus::Events::ShouldIncludeMaterialPropertyNames);
+                jobDependency.m_type = includeMaterialPropertyNames ? AssetBuilderSDK::JobDependencyType::OrderOnce : AssetBuilderSDK::JobDependencyType::Order;
```
Comment (santorac) [inline Gems/Atom/RPI/Code/Source/RPI.Builders/Model/MaterialAssetBuilderComponent.cpp:93]: We should probably have a comment to explain why we are using OrderOnce.

**Review by santorac — CHANGES_REQUESTED** [review]

File: Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp (hunk 1 above)
Comment (hershey5045) [reply Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp]: Fixed, added registry setting.

File: Gems/Atom/RPI/Code/Source/RPI.Builders/Model/MaterialAssetBuilderComponent.cpp [hunk 3]
```diff
@@ -90,7 +87,12 @@ namespace AZ
                 jobDependency.m_jobKey = "Atom Material Builder";
                 jobDependency.m_sourceFile = materialTypeSource;
                 jobDependency.m_platformIdentifier = platformIdentifier;
-                jobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
+
+                // If includeMaterialPropertyNames is true, then we need materials to depend on materialtype only once in order to get
+                // the property names. After the initial processing, materials will no longer be dependent on materialtype files.
```
Comment (santorac) [inline Gems/Atom/RPI/Code/Source/RPI.Builders/Model/MaterialAssetBuilderComponent.cpp:92]: Same here.

**Review by santorac — CHANGES_REQUESTED** [review]

File: Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp [hunk 4]
```diff
@@ -46,7 +47,7 @@ namespace AZ
         {
             AssetBuilderSDK::AssetBuilderDesc materialBuilderDescriptor;
             materialBuilderDescriptor.m_name = JobKey;
-            materialBuilderDescriptor.m_version = 107; // ATOM-14918
+            materialBuilderDescriptor.m_version = 108;
```
[the remark this answers is not in the document]
Comment (hershey5045) [reply Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp:50]: Good call, i'll leave a comment on the version update on MaterialAssetBuilderComponent as well.

**Review by santorac — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

**Pull request merged**